stock_picking_analytic/tests/test_picking_analytic_account.py

```diff
@@ -21,82 +26,81 @@ def setUpClass(cls):
                 "location_dest_id": cls.customer_location.id,
                 "picking_type_id": cls.env.ref("stock.picking_type_out").id,
                 "move_ids_without_package": [
-                    (
-                        0,
-                        0,
+                    Command.create(
                         {
                             "name": "move test",
                             "product_id": cls.product_id.id,
                             "product_uom": cls.uom_id.id,
+                            "location_id": cls.stock_location.id,
+                            "location_dest_id": cls.customer_location.id,
                         },
                     )
                 ],
             }
         )
 
-    def test_inverse_analytic_account_id(self):
-        """Set analytic account on picking
-        Check analytic account on line is set
+    def test_inverse_analytic_distribution(self):
+        """Set analytic distribution on picking
+        Check analytic distribution on line is set
         """
         picking = self.picking
-        self.assertNotEqual(len(picking.move_ids_without_package), 0)
+        self.assertTrue(picking.move_ids_without_package)
         self.assertNotEqual(
-            picking.move_ids_without_package[0].analytic_account_id,
-            self.analytic_account,
+            picking.move_ids_without_package[0].analytic_distribution,
+            self.analytic_distribution,
         )
-        picking.analytic_account_id = self.analytic_account.id
+        picking.analytic_distribution = self.analytic_distribution
         self.assertEqual(
-            picking.move_ids_without_package[0].analytic_account_id,
-            self.analytic_account,
+            picking.move_ids_without_package[0].analytic_distribution,
+            self.analytic_distribution,
         )
 
-    def test_compute_analytic_account_id(self):
-        """Set analytic account on move
-        Check analytic account on picking is set
+    def test_compute_analytic_distribution(self):
+        """Set analytic distribution on move
+        Check analytic distribution on picking is set
         """
         picking = self.picking
-        self.assertNotEqual(len(picking.move_ids_without_package), 0)
+        self.assertTrue(picking.move_ids_without_package)
         self.assertNotEqual(
-            picking.move_ids_without_package[0].analytic_account_id,
-            self.analytic_account,
+            picking.move_ids_without_package[0].analytic_distribution,
+            self.analytic_distribution,
         )
-        picking.move_ids_without_package.write(
+        picking.move_ids_without_package.update(
```

Comment:
Why update instead of write?

write is preferred, because update writes fields one by one

Same below

Reply:
`write()` would throw an error for `picking.move_ids_without_package` not having the database record when tests started.  I have switched to assign `move_ids` instead of `move_ids_without_package` to the picking in the setup to use `write()` in tests.